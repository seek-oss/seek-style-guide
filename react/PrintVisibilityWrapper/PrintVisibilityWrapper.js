import styles from './PrintVisibilityWrapper.less';
import classNames from 'classnames';

import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

export default class PrintVisibilityWrapper extends PureComponent {

  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.oneOfType([
      PropTypes.arrayOf(PropTypes.node),
      PropTypes.node
    ]).isRequired,
    isScreenOnly: PropTypes.bool,
    isPrintOnly: PropTypes.bool,
  };

  static defaultProps = {
    className: '',
    isScreenOnly: false,
    isPrintOnly: false,
  };

  render() {
    const { className, isScreenOnly, isPrintOnly, children } = this.props;

    return (
      <div
        className={classNames(
          className,
          {
            [styles.isScreenOnly]: isScreenOnly,
            [styles.isPrintOnly]: isPrintOnly,
          }
        )}
      >
        { children }
      </div>
    );
  }
}
