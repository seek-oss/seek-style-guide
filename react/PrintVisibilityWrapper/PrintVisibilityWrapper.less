.isScreenOnly {
  @media print {
    display: none;
  }
}

.isPrintOnly {
  display: none;

  @media print {
    display: block;
  }
}
