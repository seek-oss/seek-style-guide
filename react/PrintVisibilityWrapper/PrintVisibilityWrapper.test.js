import React from 'react';
import renderer from 'react-test-renderer';

import PrintVisibilityWrapper from './PrintVisibilityWrapper';

describe('PrintVisibilityWrapper:', () => {
  it('should render with children', () => {
    expect(
      renderer.create(
        <PrintVisibilityWrapper children={<div>foo</div>} />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render with className', () => {
    expect(
      renderer.create(
        <PrintVisibilityWrapper children={<div>foo</div>} className='bar' />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render with isScreenOnly', () => {
    expect(
      renderer.create(
        <PrintVisibilityWrapper children={<div>foo</div>} isScreenOnly />).toJSON()
    ).toMatchSnapshot();
  });

  it('should render with isPrintOnly', () => {
    expect(
      renderer.create(
        <PrintVisibilityWrapper children={<div>foo</div>} isPrintOnly />).toJSON()
    ).toMatchSnapshot();
  });
});
