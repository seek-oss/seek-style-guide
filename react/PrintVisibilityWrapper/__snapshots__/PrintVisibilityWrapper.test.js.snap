// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrintVisibilityWrapper: should render with children 1`] = `
<div
  className=""
>
  <div>
    foo
  </div>
</div>
`;

exports[`PrintVisibilityWrapper: should render with className 1`] = `
<div
  className="className"
>
  <div>
    foo
  </div>
</div>
`;

exports[`PrintVisibilityWrapper: should render with isPrintOnly 1`] = `
<div
  className="isPrintOnly"
>
  <div>
    foo
  </div>
</div>
`;

exports[`PrintVisibilityWrapper: should render with isScreenOnly 1`] = `
<div
  className="isScreenOnly"
>
  <div>
    foo
  </div>
</div>
`;
